syntax="proto3";

package radio_rpc;

service Radio {
    rpc GetData(Empty) returns (stream Float);
    rpc SetData(stream Float) returns (Empty);
    rpc Stop(Empty) returns (Bool);
    rpc SetParams(Parameters) returns (Bool);
}

message Empty {}

message Bool {
    bool boolean = 1;
}

message Float {
    float _float = 1;
}

message Parameters{
// TODO fill in
    float frequency = 1;
    float gain = 2;
    int32 sample_rate = 3;
}
